core/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -124,14 +126,20 @@ public BytesRef parseBytesRef(String value) {
 
         public static final String NAME = "date_time";
 
-        final FormatDateTimeFormatter formatter;
+        final String format;
         final DateTimeZone timeZone;
         private final DateMathParser parser;
+        private final DateTimeFormatter formatter;
 
         public DateTime(FormatDateTimeFormatter formatter, DateTimeZone timeZone) {
-            this.formatter = Objects.requireNonNull(formatter);
+            this.format = Objects.requireNonNull(formatter.format());
             this.timeZone = Objects.requireNonNull(timeZone);
             this.parser = new DateMathParser(formatter);
+            if (formatter.format().startsWith("epoch_millis") || formatter.format().startsWith("epoch_second")) {
+                this.formatter = formatter.printer();
```

Comment:
do we really need a special case here or could we fix the printer impl for these formats to apply the timezone when printing a date?